src/api.ts

```diff
@@ -0,0 +1,559 @@
+type ResponseTransformerFunction<IntegrationResponse, MethodResponse> = (
+  resp: IntegrationResponse
+) => MethodResponse;
+
+export abstract class BaseApiIntegration {
+  /**
+   * Identify subclasses as API integrations to the Functionless plugin
+   */
+  public static readonly FunctionlessType = "ApiIntegration";
+  protected readonly functionlessKind = BaseApiIntegration.FunctionlessType;
+
+  /**
+   * Add this integration as a Method to an API Gateway resource.
+   *
+   * TODO: this mirrors the AppsyncResolver.addResolver method, but it
+   * is on the chopping block: https://github.com/functionless/functionless/issues/137
+   * The 2 classes are conceptually similar so we should keep the DX in sync.
+   */
+  public abstract addMethod(
+    httpMethod: HttpMethod,
+    resource: aws_apigateway.Resource
+  ): void;
+}
+
+/**
+ * Static constructors for the supported API Gateway integrations.
+ * These are the preferred entrypoints as they offer superior type
+ * inference.
+ */
+export class ApiIntegrations {
+  /**
+   * Create a {@link MockApiIntegration}.
+   */
+  public static mock<
+    Request extends ApiRequest<any, any, any, any>,
+    StatusCode extends number,
+    MethodResponses extends { [C in StatusCode]: any }
+  >(
+    props: MockApiIntegrationProps<Request, StatusCode, MethodResponses>
+  ): MockApiIntegration<typeof props> {
+    return new MockApiIntegration(props);
+  }
+
+  /**
+   * Create a {@link AwsApiIntegration}.
+   */
+  public static aws<
+    Request extends ApiRequest<any, any, any, any>,
+    IntegrationResponse,
+    MethodResponse
+  >(
+    props: AwsApiIntegrationProps<Request, IntegrationResponse, MethodResponse>
+  ) {
+    return new AwsApiIntegration(props);
+  }
+}
+
+export interface MockApiIntegrationProps<
+  Request extends ApiRequest<any, any, any, any>,
+  StatusCode extends number,
+  MethodResponses extends { [C in StatusCode]: any }
+> {
+  /**
+   * Map API request to a status code. This code will be used by API Gateway
+   * to select the response to return.
+   */
+  request: RequestTransformerFunction<Request, { statusCode: StatusCode }>;
+  /**
+   * Map of status codes to response to return.
+   */
+  responses: { [C in StatusCode]: (code: C) => MethodResponses[C] };
+}
+
+/**
+ * A Mock integration lets you return preconfigured responses by status code.
+ * No backend service is invoked.
+ *
+ * To use you provide a `request` function that returns a status code from the
+ * request and a `responses` object that maps a status code to a function
+ * returning the preconfigured response for that status code. Functionless will
+ * convert these functions to VTL mapping templates and configure the necessary
+ * method responses.
+ *
+ * Only `application/json` is supported.
+ *
+ * TODO: provide example usage after api is stabilized
+ *
+ * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-api-integration-types.html
+ */
+export class MockApiIntegration<
```

Comment:
we can probably create a dummy integration for this now that returns a `{statusCode: ...}` object and use the same integration class as for AWS